string.go

```diff
@@ -57,17 +61,16 @@ func AESDecrypt(key, nonce, ciphertext []byte) ([]byte, error) {
 		return nil, err
 	}
 
-	plainText, err := gcm.Open(nil, nonce, ciphertext, nil)
-	if err != nil {
-		return nil, err
-	}
+	size := len(ciphertext) - gcm.NonceSize()
+	nonce := ciphertext[size:]
```

Comment:
You have to validate the length of the ciphertext, otherwise this is out of bounds and will panic.
Prepending the nonce makes this a bit easier?